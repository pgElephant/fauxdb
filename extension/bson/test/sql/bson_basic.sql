-- Basic BSON functionality tests
-- Test basic type creation, input/output, and simple operations

-- Create extension if not exists
CREATE EXTENSION IF NOT EXISTS bson;

-- Test basic type creation
SELECT '{}'::bson;
SELECT '[]'::bson;
SELECT '"hello"'::bson;
SELECT '123'::bson;
SELECT 'true'::bson;
SELECT 'null'::bson;

-- Test basic operators
SELECT '{"name": "test"}'::bson -> 'name';
SELECT '{"name": "test"}'::bson ->> 'name';
SELECT '{"name": "test"}'::bson ? 'name';
SELECT '{"name": "test"}'::bson ? 'missing';

-- Test array operations
SELECT '["a", "b", "c"]'::bson -> '0';
SELECT '["a", "b", "c"]'::bson ->> '0';

-- Test nested objects
SELECT '{"user": {"name": "john", "age": 30}}'::bson -> 'user' -> 'name';
SELECT '{"user": {"name": "john", "age": 30}}'::bson -> 'user' ->> 'age';

-- Test comparison operators
SELECT '{"a": 1}'::bson = '{"a": 1}'::bson;
SELECT '{"a": 1}'::bson != '{"a": 2}'::bson;

-- Test containment operators
SELECT '{"a": 1, "b": 2}'::bson @> '{"a": 1}'::bson;
SELECT '{"a": 1}'::bson <@ '{"a": 1, "b": 2}'::bson;

-- Test existence operators
SELECT '{"a": 1, "b": 2}'::bson ?| ARRAY['a', 'c'];
SELECT '{"a": 1, "b": 2}'::bson ?& ARRAY['a', 'b'];
