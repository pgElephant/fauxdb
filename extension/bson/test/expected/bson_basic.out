-- Basic BSON functionality tests
-- Test basic type creation, input/output, and simple operations

-- Create extension
CREATE EXTENSION

-- Test basic type creation
 ?column? 
----------
 {}
(1 row)

 ?column? 
----------
 []
(1 row)

 ?column? 
----------
 "hello"
(1 row)

 ?column? 
----------
 123
(1 row)

 ?column? 
----------
 true
(1 row)

 ?column? 
----------
 null
(1 row)

-- Test basic operators
 ?column? 
----------
 "test"
(1 row)

 ?column? 
----------
 test
(1 row)

 ?column? 
----------
 t
(1 row)

 ?column? 
----------
 f
(1 row)

-- Test array operations
 ?column? 
----------
 "a"
(1 row)

 ?column? 
----------
 a
(1 row)

-- Test nested objects
 ?column? 
----------
 "john"
(1 row)

 ?column? 
----------
 30
(1 row)

-- Test comparison operators
 ?column? 
----------
 t
(1 row)

 ?column? 
----------
 t
(1 row)

-- Test containment operators
 ?column? 
----------
 t
(1 row)

 ?column? 
----------
 t
(1 row)

-- Test existence operators
 ?column? 
----------
 t
(1 row)

 ?column? 
----------
 t
(1 row)
